app/models/manageiq/providers/embedded_ansible/automation_manager/job.rb

```diff
@@ -3,8 +3,100 @@ class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Job < ManageIQ::P
 
   require_nested :Status
 
+  def self.run_playbook(task_id)
+    task = MiqTask.find(task_id)
+    options = task.context_data
+    name = build_name(options[:name])
+    description = options[:description]
+    info = options[:config_info]
+    playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+    tower = playbook.manager
+    info[:inventory] = create_and_return_inventory(tower, name, description, task_id) if info[:inventory].nil?
+    _tower, params = ServiceTemplateAnsiblePlaybook.build_parameter_list(name, description, info)
+    job = nil
+    tower.with_provider_connection do |connection|
+      # Launch shouldn't need options since extra_vars can be included in the creation
+      #   of the job template
+      job = connection.api.job_templates.create!(params).launch
+      message("Created Job Template and Launched", task_id)
+      job
+    end
+    requeue_job_status(job.id, task_id)
+  end
+
+  def self.message(msg, task_id)
+    task = MiqTask.find(task_id)
+    task.update_message(msg)
+  end
+
+  def self.reload_job_status(job_id, options)
+    info = options[:config_info]
+    playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+    tower = playbook.manager
+    tower.with_provider_connection do |connection|
+      connection.api.jobs.find(job_id)
+    end
+  end
+
+  def self.requeue_job_status(job_id, task_id)
+    task = MiqTask.find(task_id)
+    options = task.context_data
+    job = reload_job_status(job_id, options)
+    if job.status == 'successful' || job.status == 'failed'
+      # Modify the task to include the error on fail
+      task.update_status("Finished", "Ok", "Playbook launch finished successfully") if task
+    else
+      task.update_status("Active", "Ok", "Playbook still running") if task
+      info = options[:config_info]
+      playbook = ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Playbook.find(info[:playbook_id])
+      ems = playbook.manager
+      cb = {:class_name => task.class.name, :instance_id => task.id, :method_name => :queue_callback_on_exceptions, :args => ['Finished']}
+      MiqQueue.put(
+        :class_name  => "ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Job",
+        :method_name => 'requeue_job_status',
+        :role        => 'embedded_ansible',
+        :args         => [job.id, task.id],
+        :miq_callback => cb,
+        :zone         => ems.my_zone
```

Comment:
@syncrou Need to set :task_id to nil here to break the task inheritance or this task will block.
https://github.com/ManageIQ/manageiq/blob/cbc28597c7def05a79e1f3c28bff4e7bb287bd39/app/models/ems_refresh.rb#L75